### PR TITLE
Fix: LH Unused CSS + Organizes CMS Sections and components styles.

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

- [x] Enables `SlideOver`/`Modal` components to be used as Sections (`CartSidebar`, `NavbarSlider`, `FilterSlider`,`RegionModal`).
- [x] Organizes `Sections`/`Common` components into Layout and Pages.
- [x] Uses CSS Modules per Sections (3 styles customizations levels, see video in slack channel).
- [x] Removes CSS Modules from other components and move styles to `@faststore/ui`.
- [x] Adjusts styles to use the same pattern used in the Atomic Design Epic approach in `@faststore/ui`.
- [x] Removes some stories (Storybook).  
- [x] Fix Unused CSS Problem for Homepage, PDP, PLP. 
- [x] Fix conflicts.

## How it works?

It prepares the repo to uses the CMS Section approach. It also enables the CSS Modules per section instead of importing all styles globally as before. Finally it fixes the Unused CSS Problem for Homepage, PDP, PLP, Search etc.

## How to test it?

All the pages should keep the same visually in Desktop and Mobile versions. You can see the coverage tab in devtools and check the unused CSS compared to previous version using the starter PR.

Note that the unused CSS problem is gone in the LH Check from the starter PR (But there are others LH Issues yet).

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/44

### CODE REVIEW

**Review by ArthurTriis1 — APPROVED** [review]
LGTM! I found just a margin in the footer, but will pre-approve!
Screenshot 2023-04-20 at 16 44 56

**Comment by hellofanny** [comment]
> LGTM! I found just a margin in the footer, but will pre-approve!
> Screenshot 2023-04-20 at 16 44 56

Ya! Same think for desktop version
image

File: packages/components/src/molecules/Modal/Modal.tsx [hunk 1]
```diff
@@ -38,6 +38,10 @@ export interface ModalProps extends Omit<ModalContentProps, 'children'> {
    * Event emitted when the modal is closed
    */
   onDismiss?: () => void
+  /**
+   * Props forwarded to overlay component
```
Comment (hellofanny) [inline packages/components/src/molecules/Modal/Modal.tsx:42]:
```suggestion
   * Props forwarded to the `Overlay` component
```

File: packages/components/src/organisms/CartSidebar/CartSidebar.tsx [hunk 2]
```diff
@@ -58,6 +63,8 @@ function CartSidebar({
   alertIcon,
   alertText,
   onClose,
+  overlayProps,
+  ...otherProps
```
Comment (hellofanny) [inline packages/components/src/organisms/CartSidebar/CartSidebar.tsx:67]:
```suggestion
  overlayProps,
  onClose,
  ...otherProps
```

File: packages/ui/src/styles/global.scss [hunk 3]
```diff
@@ -1,2 +1 @@
 @import "./base";
-@import "./components";
```
Comment (hellofanny) [inline packages/ui/src/styles/global.scss:2]: With these changes, [this](https://faststore-site-git-feat-fs-778-creates-navbar-71017f-faststore.vercel.app/components#all-components-included) won't be true anymore, right?

image

So, we should change to 

```
All components included
@import "@faststore/ui/src/styles/components.scss";
```

Is that right?

**Review by hellofanny — APPROVED** [review]
Excellent job forms 🚀 🌟 !!! Thanks for taking time to improve this issue, it's a critical PR and changes a lot of files.

I've tested through the application and everything seems fine but the footer margin mentioned above 🤩

File: packages/components/src/organisms/SlideOver/SlideOver.tsx [hunk 4]
```diff
@@ -44,6 +48,7 @@ function SlideOver({
   children,
   onDismiss,
   testId = 'fs-slide-over',
+  overlayProps,
```
Comment (lucasfp13) [inline packages/components/src/organisms/SlideOver/SlideOver.tsx:51]:
```suggestion
  overlayProps,
  testId = 'fs-slide-over',
```

**Review by lucasfp13 — APPROVED** [review]
LGTM

File: packages/components/src/organisms/CartSidebar/CartSidebar.tsx [hunk 5]
```diff
@@ -46,6 +47,10 @@ export interface CartSidebarProps {
    * Function called when Close Button is clicked.
    */
   onClose: () => void
+  /**
+   * Props forwarded to overlay component
+   */
+  overlayProps?: OverlayProps
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/components/src/organisms/CartSidebar/CartSidebar.tsx]: nice! done [here](https://github.com/vtex/faststore/pull/1712/commits/29026696e67c54061b1a7e724a68902cd688baed).

File: packages/components/src/organisms/CartSidebar/CartSidebar.tsx (hunk 2 above)
Comment (eduardoformiga) [reply packages/components/src/organisms/CartSidebar/CartSidebar.tsx]: nice! done [here](https://github.com/vtex/faststore/pull/1712/commits/29026696e67c54061b1a7e724a68902cd688baed).

File: packages/components/src/organisms/Filter/FilterSlider.tsx [hunk 6]
```diff
@@ -1,8 +1,10 @@
+import type { HTMLAttributes } from 'react'
 import React, { PropsWithChildren } from 'react'
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/components/src/organisms/Filter/FilterSlider.tsx]: Nice! done.

File: packages/components/src/organisms/Filter/FilterSlider.tsx [hunk 7]
```diff
@@ -36,6 +38,10 @@ export interface FilterSliderProps {
    * Function called when Close Button is clicked.
    */
   onClose: () => void
+  /**
+   * Props forwarded to overlay component
+   */
+  overlayProps?: OverlayProps
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/components/src/organisms/Filter/FilterSlider.tsx]: nice! done [here](https://github.com/vtex/faststore/pull/1712/commits/29026696e67c54061b1a7e724a68902cd688baed).

File: packages/ui/src/styles/global.scss (hunk 3 above)
Comment (eduardoformiga) [reply packages/ui/src/styles/global.scss:2]: Nice catch! Makes sense to me. Done [here](https://github.com/vtex/faststore/pull/1712/commits/986405809f9e487bb56331a321f52577a5970bb0).

**Comment by eduardoformiga** [comment]
@ArthurTriis1  @hellofanny what do you think about [this change](https://github.com/vtex/faststore/pull/1712/commits/5e8c316a6681e8db1d9a58b7c277616af98ea8aa) to fix the footer padding? I wonder if would be better place some styles in the `data-fs-footer` instead.

**Comment by ArthurTriis1** [comment]
> @ArthurTriis1 @hellofanny what do you think about [this change](https://github.com/vtex/faststore/pull/1712/commits/5e8c316a6681e8db1d9a58b7c277616af98ea8aa) to fix the footer padding? I wonder if would be better place some styles in the `data-fs-footer` instead.

I think we can do this, and when the footer is moved, we put these styles in data-attributes

**Review by hellofanny — CHANGES_REQUESTED** [review]
@eduardoformiga excellent work!!!

I've checked the documentation [preview](https://faststore-site-git-fix-unused-css-lh-faststore.vercel.app/components/molecules/quantity-selector), and all the components styles are missing.

we forgot to update [apps/site/pages/_app.tsx ](https://github.com/vtex/faststore/blob/main/apps/site/pages/_app.tsx). 
We should import `@faststore/ui/src/styles/components.scss` there too! Can u add it pls, and check it again? Thanks ⭐ 

image

**Comment by eduardoformiga** [comment]
> 58034cf

Nice catch, Fan.

I changed [it here](https://github.com/vtex/faststore/pull/1712/commits/58034cf53d2ee9bf03b24a387af47931c84ed150). Can you double check it again?

File: apps/site/pages/components/molecules/navbar-slider.mdx [hunk 8]
```diff
@@ -67,7 +67,7 @@ import {
 Import Styles
 
 ```tsx
-import '@faststore/ui/src/components/molecules/NavbarSlider/styles.scss'
```
Comment (hellofanny) [inline apps/site/pages/components/molecules/navbar-slider.mdx:70]: nice catch!

**Review by hellofanny — APPROVED** [review]
Reviewed again and LGTM!!! 🚀 

Everything seems to work as expected 🌟 
image

**Pull request merged**